drivers/storage/portworx/manifest/configmap.go

```diff
@@ -0,0 +1,54 @@
+package manifest
+
+import (
+	"context"
+
+	corev1alpha1 "github.com/libopenstorage/operator/pkg/apis/core/v1alpha1"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultConfigMapName = "px-versions"
+	versionConfigMapKey  = "versions"
+)
+
+type configMap struct {
+	cm *v1.ConfigMap
+}
+
+func newConfigMapManifest(
+	k8sClient client.Client,
+	cluster *corev1alpha1.StorageCluster,
+) (manifest, error) {
+	versionCM := &v1.ConfigMap{}
+	err := k8sClient.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      defaultConfigMapName,
+			Namespace: cluster.Namespace,
+		},
+		versionCM,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &configMap{
+		cm: versionCM.DeepCopy(),
+	}, nil
+}
+
+func (m *configMap) Get() (*Version, error) {
+	data, exists := m.cm.Data[versionConfigMapKey]
+	if !exists {
```

Comment:
Not a strong reason to not error out.
Most people would download the versions file and create a config map using that. With kubectl the file name becomes the key in the configmap.
Ideally if users follow the instructions it should have the correct file name and thus key in the configmap.
Just keep this check lenient in case people use wrong file name. Also in case of failure we would end of using default versions with a warning. I thought it would be better to use whatever we have then use defaults.